add modal for component information

mmtywa opened the pull request:
Media break point at 987px. Def check what it looks like at 986px.

Questions:
Product Image Section:

- Are all products recommended (need the recommended picture/badge)?
- Space before or after the product image? (see image below)

![Screen Shot 2023-04-13 at 17 40 58](https://user-images.githubusercontent.com/129784422/231914445-f5006aed-59b7-4415-95a2-9b897246f63a.png)

2.00b Staff Help Section:

- #ask-the-tas always an option?
- Case for when no staff to help? Display a message maybe?

mmtywa committed:
add modal for component information
- created modal component and styles
- added modal root to electronic-library.tsx
- added inVault and helpful staff properties to component object and interface

poofytoo commented:
Ah this is great! I'll just leave responses to your notes here:

**Product Image Section:**
Are all products recommended (need the recommended picture/badge)?
**Yes**. All products are recommended and should use that badge :) (I know, silly, right?). One other thing is the image should be unstretched, please!)
Space before or after the product image? (see image below)
**Yes**. There should be maybe ~15px of space above and below the image when it's in mobile.
**When it's not, there should be the same amount of space to the left of the image, as there is around the 'modal card'**

2.00b Staff Help Section:
**Yes** Let's put #ask-the-tas as always an option. Also, my bad, let's change the text to: `You reach out to the following staff for help using this component!`

The Close X font should be DM Sans. Thank you for using the nice exit X instead of just a plaintext X!! This is super important to me <3 

Can you make sure that the 4 green lines are all the same space (padding/margin etc.)
![image](https://user-images.githubusercontent.com/276204/231931366-cab3da14-7a57-4cd4-9f64-5ee9bc5cd120.png)

Sorry, I know this is an outdated review request and was probably my fault, but can you change this:
![image](https://user-images.githubusercontent.com/276204/231931710-20a15856-8503-472a-b75d-5cbea5ba47f6.png)
The search bar should be 100%, with a max-width of 460px; That way it won't break the mobile page (as it does currently)

Again - this is definitely a result of my poorly spec'd mocks. But when the screen is large, don't space out the text evenly (red arrows) in general, text should always flow one after the other when it's vertical, with the same amount of spacing in between paragraphs/sections :) 

If the text doesn't fill the whole space, then the white box shrinks. (blue arrows) It's centered inside this 'invisible box'. 

The space above the close button and to the right of it is still the same (green lines). Happy to clarify about all this - just ping me
![image](https://user-images.githubusercontent.com/276204/231932309-86be1395-e8b5-47f3-8416-3bece727f388.png)

This was super fast, and a great first pass! Thank you!!

mmtywa committed:
adds revisions from poofytoo review
- fixes recommended product badge stretching
- adds space around image
- Staff Help: ask-the-tas always an option
- fixes font on close X
- 4 green lines equal: equal spacing top,right,bottom of close and right,bottom,left of card
- fixes width of search bar

poofytoo commented:
I think on mobile, some of the content don't shrink down correctly (mainly the lead time container)

<img width="410" alt="image" src="https://user-images.githubusercontent.com/276204/232097270-41c2872f-bd16-461d-9f1e-a0cedf4af7f1.png">

Can you reduce (on mobile, probably a media query) the spacing between the edge of the phone screen and the white box, and then also the white box to the content? Maybe something closer to 20px?

Could you add a slight round (10px) to the images? (sorry, this wasn't in the mocks)
![image](https://user-images.githubusercontent.com/276204/232097697-7bf15958-f324-4734-be05-60c88e8c1add.png)

Visually, those are the only things I have left I think! I'll do a quick code pass now (normally Dillon would do this but I know he's volunteering today)

mmtywa committed:
adds round 2 revisions from poofytoo
- fixes spacing on mobile
- adds media break points at 400px and 720px
- adds border radius 10px to product image

mmtywa committed:
adds max width to web url
- fixes text overflow below 400px device width

poofytoo commented on components/Modal/Modal.module.scss:
```diff
@@ -0,0 +1,277 @@
+      margin: 0;
+    }
+
+    span {
```
heyo - in general, please avoid tag/type/dangling element selectors (https://medium.com/@reemtos/dangling-element-selectors-in-css-and-why-you-should-avoid-them-171f6cd989b1) 

certain ones are more fine if they're a little bit semantic - i.e. `em` or `strong` as they describe emphasis or the strong tag, but `span` is just very generic (and if I did that in the past whoops you can blame me haha). but if possible try to come up with semantic classnames

mmtywa replied:
Awesome! Thanks for the resource. I left a couple in there as resets for anchor and p tags, otherwise I got it pretty cleaned up.

poofytoo commented on components/Modal/index.tsx:
```diff
@@ -0,0 +1,180 @@
+  } else if (link?.toLowerCase().search(/\S*adafruit\S*/) !== -1) {
+    return 'Adafruit Website';
+  } else {
+    return link ? link : 'missing link :(';
```
Instead of saying missing link, can you just say `check with a TA`

poofytoo commented on components/Modal/index.tsx:
```diff
@@ -0,0 +1,180 @@
+const buildLinkText = (link?: string): string => {
+  if (link?.toLowerCase().search(/\S*arduino\S*/) !== -1) {
+    return 'Arduino Website';
+  } else if (link?.toLowerCase().search(/\S*adafruit\S*/) !== -1) {
```
ooo perfect. can you also add one for Grainger `Grainger website`

poofytoo commented on components/Modal/index.tsx:
```diff
@@ -0,0 +1,180 @@
+  useEffect(() => {
+    setIsBrowser(true);
+    const close = (e: KeyboardEvent) => {
+      if (e.key === 'Escape') {
```
nice!

poofytoo reviewed (commented):
A couple of comments, but also this fella: 

<img width="1832" alt="image" src="https://user-images.githubusercontent.com/276204/232140144-7023d918-1ec3-440d-bcc9-cd263d21cb2d.png">

I think that 'actuation' is supposed to have a black background, but for some reason it's showing up as white so the tag just looks like a giant space. - this should just be a default one, if there are any undefined tags it's just white on a black background tag (you probably already know this but just in case...)

mmtywa committed:
adds round 3 code cleanup
- cleans up scss dangling element selectors
- changes message for no purchase link
- adds regex for Grainger
